mafContrib/TDEsPopMetric.py

```diff
@@ -0,0 +1,125 @@
+            self.data.append(new_dict)
+
+    def interp(self, t, filtername, lc_indx=0):
+        """
+        t : array of floats
+            The times to interpolate the light curve to.
+        filtername : str
+            The filter. one of ugrizy
+        lc_index : int (0)
+            Which file to use.
+        """
+
+        result = np.interp(t, self.data[lc_indx][filtername]['ph'],
+                           self.data[lc_indx][filtername]['mag'],
+                           left=99, right=99)
+        return result
+
+
+class TDEsPopMetric(metrics.BaseMetric):
+    def __init__(self, metricName='TDEsPopMetric', mjdCol='observationStartMJD', m5Col='fiveSigmaDepth',
+                 filterCol='filter', nightCol='night', ptsNeeded=2, file_list=None, mjd0=59853.5,
+                 **kwargs):
+        maps = ['DustMap']
+        self.mjdCol = mjdCol
+        self.m5Col = m5Col
+        self.filterCol = filterCol
+        self.nightCol = nightCol
+        self.ptsNeeded = ptsNeeded
+
+        self.lightcurves = tde_lc(file_list=file_list)
+        self.mjd0 = mjd0
+
+        waveMins = {'u': 330., 'g': 403., 'r': 552., 'i': 691., 'z': 818., 'y': 950.}
+        waveMaxes = {'u': 403., 'g': 552., 'r': 691., 'i': 818., 'z': 922., 'y': 1070.}
+
+        self.a = {}
+        self.b = {}
+        for filtername in waveMins.keys():
+            testsed = Sed()
+            testsed.setFlatSED(wavelen_min=waveMins[filtername],
+                               wavelen_max=waveMaxes[filtername],
+                               wavelen_step=1)
+            self.a[filtername], self.b[filtername] = testsed.setupCCM_ab()
+        self.R_v = 3.1
+
+        cols = [self.mjdCol, self.m5Col, self.filterCol, self.nightCol]
+        super(TDEsPopMetric, self).__init__(col=cols, units='Detected, 0 or 1',
+                                            metricName=metricName, maps=maps,
+                                            **kwargs)
+
+    def run(self, dataSlice, slicePoint=None):
+        result = 0
+        t = dataSlice[self.mjdCol] - self.mjd0 - slicePoint['peak_time']
+        mags = np.zeros(t.size, dtype=float)
+
+        for filtername in np.unique(dataSlice[self.filterCol]):
+            infilt = np.where(dataSlice[self.filterCol] == filtername)
+            mags[infilt] = self.lightcurves.interp(t[infilt], filtername, lc_indx=slicePoint['file_indx'])
+            # Apply dust extinction on the light curve
+            A_x = (self.a[filtername][0]+self.b[filtername][0]/self.R_v)*(self.R_v*slicePoint['ebv'])
+            mags[infilt] -= A_x
+
+        # Simple alert criteria. Could make more in depth, or use reduce functions
+        # to have multiple criteria checked.
+        pre_peak_detected = np.where((t < 0) & (mags < dataSlice[self.m5Col]))[0]
+
+        if pre_peak_detected.size > self.ptsNeeded:
+            result = 1
+
+        return result
+
+
+def TDEsPopSlicer(t_start=1, t_end=3652, n_events=10000, seed=42, n_files=7):
+
```

Comment:
Can you add a comment to make it easier to understand what this does? It seems like the point is to set up a random location for the TDE events over the sphere, and then assign random times to when the TDEs occur, so that each slicepoint then knows not only where it is but when the TDE should be occuring. 

Should there be a parent class TransientPopSlicer which does this but has something like 

```
class TransientPopSlicer) 
      def init( ??   number of points?): 
           self.setSpatialPoints()
           self.setTransientTimes()
     def setSpatialPoints():
          default set up of uniform sphere with the UserPointsSlicer (which could then be overriden by child classes)
      def setTransientTimes(): 
          maybe default is completely random times for events (but obviously overrideable and maybe most likely to be)
```

and could even make child classes TransientGalacticSlicer where the default uniform sphere is replaced by something dependent on the stellar density ?

Reply:
Also .. this is a function, not a class -- I read it as a class (because it's capitalized). If this is going to be a function, probably could rename as "setupTdePopSlicer" so it's more clear to someone reading fast?